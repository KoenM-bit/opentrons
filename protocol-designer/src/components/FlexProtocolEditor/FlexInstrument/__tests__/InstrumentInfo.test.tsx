import * as React from 'react'
import { shallow } from 'enzyme'
import { InstrumentInfo, InstrumentInfoProps } from '../InstrumentInfo'

describe('InstrumentInfoComponent', () => {
  it('renders correctly with all props provided', () => {
    const props: InstrumentInfoProps = {
      mount: 'left',
      isDisabled: false,
      className: 'test-class',
      pipetteSpecs: {
        displayCategory: 'GEN2',
        channels: 1,
      },
      infoClassName: 'test-info-class',
      showMountLabel: true,
      description: 'This is a pipette',
      tiprackModel: 'opentrons_96_tiprack_300ul',
      children: <div>Test Child Component</div>,
    }

    const wrapper = shallow(<InstrumentInfo {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with minimum required props provided', () => {
    const props: InstrumentInfoProps = {
      mount: 'left',
      description: 'This is a pipette',
      isDisabled: false,
    }

    const wrapper = shallow(<InstrumentInfo {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with long description', () => {
    const props: InstrumentInfoProps = {
      mount: 'left',
      description:
        'This is a very long description that should still render correctly even though it takes up multiple lines and is quite verbose.',
      isDisabled: false,
    }

    const wrapper = shallow(<InstrumentInfo {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
