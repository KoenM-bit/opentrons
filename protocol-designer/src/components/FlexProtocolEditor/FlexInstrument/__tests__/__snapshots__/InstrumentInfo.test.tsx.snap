// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentInfoComponent renders correctly with all props provided 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Card>
    <div
      className="pipette left test-class"
    >
      <InstrumentDiagram
        className="pipette_icon"
        mount="left"
        pipetteSpecs={
          Object {
            "channels": 1,
            "displayCategory": "GEN2",
          }
        }
      />
      <div
        className="pipette_info test-info-class"
      >
        <InfoItem
          title="left pipette"
          value="This is a pipette"
        />
        <InfoItem
          title="tip rack"
          value="opentrons_96_tiprack_300ul"
        />
        <div>
          Test Child Component
        </div>
      </div>
    </div>
  </Card>
</div>
`;

exports[`InstrumentInfoComponent renders correctly with long description 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Card>
    <div
      className="pipette left"
    >
      <InstrumentDiagram
        className="pipette_icon"
        mount="left"
      />
      <div
        className="pipette_info"
      >
        <InfoItem
          title="pipette"
          value="This is a very long description that should still render correctly even though it takes up multiple lines and is quite verbose."
        />
      </div>
    </div>
  </Card>
</div>
`;

exports[`InstrumentInfoComponent renders correctly with minimum required props provided 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Card>
    <div
      className="pipette left"
    >
      <InstrumentDiagram
        className="pipette_icon"
        mount="left"
      />
      <div
        className="pipette_info"
      >
        <InfoItem
          title="pipette"
          value="This is a pipette"
        />
      </div>
    </div>
  </Card>
</div>
`;
